src/v/cloud_storage/types.cc

```diff
@@ -262,17 +267,53 @@ size_t anomalies::count_segment_meta_anomaly_type(anomaly_type type) const {
     return static_cast<size_t>(count);
 }
 
+/// Returns number of discarded elements
+template<class T, size_t quota = max_number_of_manifest_anomalies>
```

Comment:
nit: IMO `size_limit` would be more descriptive than `quota`

Reply:
fixed